markdown_it/token.py

```diff
@@ -15,8 +37,10 @@ class Token:
     # -  `0` means the tag is self-closing
     # - `-1` means the tag is closing
     nesting: int = attr.ib()
-    # Html attributes. Format: `[ [ name1, value1 ], [ name2, value2 ] ]`
-    attrs: Optional[List[list]] = attr.ib(default=None)
+    # Html attributes. Note this differs from the upstream "list of lists" format
+    attrs: Dict[str, Union[str, int, float]] = attr.ib(
```

Comment:
Actually would be less painful to just use `Any` instead of the `Union[str, float]]` in case someone gets an item directly without `attrGet`. I've added suggestions also elsewhere to annotate read API with `Any` and write API with `Union[str, float]` because of this.

Reply:
I'm a bit confused why you suggest this, because `isinstance(1, float)` equates to `False` so int are certainly not a subclass of float.
As you mention, `int` is used in core otherwise really I was thinking that perhaps it could just be `str` only, since all these values always anyway have to be converted to strings in HTML attributes `<div key="value">`.
I can't think of any reason, or HTML tag, that would require it to not to be one of `str`, `int`, or `float` and this restriction ensures that `Token` is always "JSONable"

Reply:
This is in PEP 484 (https://www.python.org/dev/peps/pep-0484/#the-numeric-tower):
> when an argument is annotated as having type float, an argument of type int is acceptable

The argument typing system is not one to one with `isinstance`, it is more "duck typed".

> it could just be str

I like this idea but only if we take the liberty to diverge from upstream and actually always set `str` values only, i.e. convert before assigning.

Reply:
> The argument typing system is not one to one with isinstance, it is more "duck typed".

Fair, although I think it is better for human readability/understanding to include `int` specifically. Bit weird IMO, that typing should diverge from isinstance

> but only if we take the liberty to diverge from upstream

yeh thats why I ultimately decided against it, just not worth the hassle